zerver/views/users.py

```diff
@@ -262,6 +341,9 @@ def update_user_backend(
                 )
         validate_user_custom_profile_data(target.realm.id, clean_profile_data)
         do_update_user_custom_profile_data_if_changed(target, clean_profile_data)
+        notify_users_on_updated_user_fields(
+            user_profile, target, target.profile_data(), profile_data
+        )
```

Comment:
`notify_users_on_updated_user_fields` should run before` do_update_user_custom_profile_data_if_changed` so that we can still retrive the `old_profile_data` before it is updated with the new one